kyaml/kio/byteio_reader.go

```diff
@@ -101,6 +101,9 @@ type ByteReader struct {
 	// WrappingKind is set by Read(), and is the kind of the object that
 	// the read objects were originally wrapped in.
 	WrappingKind string
+
+	// Path indicates file path which is being read
+	Path string
```

Comment:
Probably want to use `SerializationMode` or something rather than the path.

Reply:
I feel Path is more handy as it is the source of truth for the file ext and SerializationMode/DeserializationMode

Reply:
What should path be if reading json from stdin?

Reply:
@pwittrock It would be empty. Updated the comment.

Should we accept json input in stdin as well ? If so, we should parse the json prior to determine if it is json. The issue filed is for accepting json format file. Let me know your thoughts.

Reply:
The ByteReader shouldn't need to know or care about what the source of the bytes were -- whether it was a file, stdin, http request, etc.  Doing so breaks the abstraction -- it reads bytes.  If we would to make an input source that read from the apiserver, it would support reading json and use the ByteReader to parse the response bytes.

The format should also probably be an annotation that is set on the resource when it is read -- similar to how the index is an annotation on the resource.  This way the format would be preserved across processes.

Reply:
Agree. Added AcceptJSON flag instead. PTAL